cdap-messaging-ext-spanner/src/main/java/io/cdap/cdap/messaging/spanner/SpannerMessagingService.java

```diff
@@ -0,0 +1,303 @@
+        this.instanceId, this.databaseId, ddlStatements, null);
+    try {
+      future.get();
+    } catch (InterruptedException | ExecutionException e) {
+      LOG.error("Failed to create topic {}", topicMetadata.getTopicId().getTopic(), e);
+      throw new IOException(e);
+    }
+
+    Gson gson = new Gson();
+    String jsonString = gson.toJson(topicMetadata.getProperties());
+    Mutation mutation = Mutation.newInsertOrUpdateBuilder(TOPIC_METADATA_TABLE).set(TOPIC_ID_FIELD)
+        .to(getTableName(topicMetadata.getTopicId())).set(PROPERTIES_FIELD)
+        .to(Value.json(jsonString)).set(NAMESPACE_FIELD)
+        .to(topicMetadata.getTopicId().getNamespace()).build();
+    try {
+      client.write(Collections.singleton(mutation));
+    } catch (SpannerException e) {
+      LOG.error("Failed to update topic metadata table for {}",
+          topicMetadata.getTopicId().getTopic(), e);
+      throw new IOException(e);
+    }
+    LOG.info("Created topic : {}", topicMetadata.getTopicId().getTopic());
+  }
+
+  private static String getCreateTopicMetadataDDLStatement() {
+    return String.format(
+        "CREATE TABLE IF NOT EXISTS %s ( %s STRING(MAX) NOT NULL, %s STRING(MAX), %s JSON ) PRIMARY KEY(%s)",
+        TOPIC_METADATA_TABLE, TOPIC_ID_FIELD, NAMESPACE_FIELD, PROPERTIES_FIELD, TOPIC_ID_FIELD);
+  }
+
+  private static String getCreateTopicDDLStatement(TopicId topicId) {
+    return String.format("CREATE TABLE IF NOT EXISTS %s ( %s INT64, %s INT64, %s"
+            + " TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true), %s BYTES(MAX) )"
+            + " PRIMARY KEY (%s, %s, %s), ROW DELETION POLICY"
+            + " (OLDER_THAN(publish_ts, INTERVAL 7 DAY))", getTableName(topicId), SEQUENCE_ID_FIELD,
+        PAYLOAD_SEQUENCE_ID, PUBLISH_TS_FIELD, PAYLOAD_FIELD, SEQUENCE_ID_FIELD,
+        PAYLOAD_SEQUENCE_ID, PUBLISH_TS_FIELD);
+  }
+
+  public static String getTableName(TopicId topicId) {
+    return topicId.getNamespace() + topicId.getTopic();
+  }
+
+  @Override
+  public void updateTopic(TopicMetadata topicMetadata)
+      throws TopicNotFoundException, IOException, UnauthorizedException {
+
+    String topicId = topicMetadata.getTopicId().getTopic();
+    Gson gson = new Gson();
+    String jsonString = gson.toJson(topicMetadata.getProperties());
+
+    // Update the topic properties in the TopicMetadata table
+    Mutation mutation = Mutation.newUpdateBuilder(TOPIC_METADATA_TABLE).set(TOPIC_ID_FIELD)
+        .to(topicId).set(PROPERTIES_FIELD).to(Value.json(jsonString)).build();
+
+    try {
+      client.write(Collections.singleton(mutation));
+    } catch (SpannerException e) {
+      if (e.getErrorCode() == ErrorCode.NOT_FOUND) {
+        throw new TopicNotFoundException(topicMetadata.getTopicId().getNamespace(),
+            topicMetadata.getTopicId().getTopic());
+      }
+      LOG.error("Failed to update topic {}", topicId, e);
```

Comment:
It's not a good practice to both log and throw an exception. I'd only throw an exception here and let client handles the exception.